packages/expo-apple-authentication/src/AppleAuthentication.types.ts

```diff
@@ -0,0 +1,284 @@
+*/
+export type AppleAuthenticationCredential = {
+  /**
+   * A value indicating the status type of the requested credential.
+   * Success if the credential was retrieved successfully,
+   * Revoke if the credential was revoked,
+   * or Cancel if the user canceled the Sign In operation.
+   */
+  type: AppleAuthenticationStatus;
+
+  /**
+    * A JSON Web Token (JWT) that securely communicates information about the user to your app.
+    */
+  identityToken?: string;
+
+  /**
+    * 	A short-lived token used by your app for proof of authorization when interacting with the app’s server counterpart.
+    */
+  authorizationCode?: string;
+
+  /**
+    * An arbitrary string that your app provided to the request that generated the credential.
+    * You can set this in `RequestOptions`.
+    */
+  user?: string;
+
+  /**
+    * An identifier associated with the authenticated user.
+    * You can use this to check if the user is still authenticated later.
+    * This is stable and can be shared across apps released under the same development team.
+    * The same user will have a different identifier for apps released by other developers.
+    */
+  state?: string;
+
+  /**
+    * The contact information the user authorized your app to access.
+    */
+  authorizedScopes?: AppleAuthenticationScope[];
+
+  /**
+    * The user’s name. Might not present if you didn't request access or if the user denied access.
+    */
+  fullName?: AppleAuthenticationFullName;
+
+  /**
+    * The user’s email address. Might not present if you didn't request access or if the user denied access.
+    */
+  email?: string;
+
+  /**
+    * A value that indicates whether the user appears to be a real person.
+    */
+  realUserStatus?: AppleAuthenticationUserDetectionStatus;
+}
+
+/**
+ * Tokenized object representing the different portions of the user's full name.
+ */
+export type AppleAuthenticationFullName = {
+  namePrefix?: string;
+  givenName?: string;
+  middleName?: string;
+  familyName?: string;
+  nameSuffix?: string;
+  nickname?: string;
+}
+
+/**
+ * Event sent to the listener when the user's credentials have been revoked.
+ */
+export type AppleAuthenticationRevokeEvent = {
+  type: AppleAuthenticationStatus;
+}
+
+/**
+ * Listener that is called when the user's credentials have been revoked.
+ */
+export type AppleAuthenticationRevokeListener = (event: AppleAuthenticationRevokeEvent) => void;
+
+/**
+ * Controls which scopes you are requesting when the call `AppleAuthentication.requestAsync()`.
+ *
+ * @note Note that it is possible that you will not be granted all of the scopes which you request.
+ * You need to check which ones you are granted in the `Credential` you get back.
+ *
+ * @see [Apple documention](https://developer.apple.com/documentation/authenticationservices/asauthorizationscope) for more details.
+ */
+export enum AppleAuthenticationScope {
+  /**
+   * A scope that includes the user’s full name.
+   */
+  FULL_NAME = 0,
```

Comment:
Shouldn't TypeScript enums use `UpperCamelCase` instead of `ALL_CAPS_SNAKE_CASE`?

Reply:
We use all uppercase for Expo.